runtime/runtime-params-estimator/src/vm_estimator.rs

```diff
@@ -89,26 +96,23 @@ pub fn cost_per_op(gas_metric: GasMetric, code: &ContractCode) -> Ratio<u64> {
     // to the notion of 1M gas as nanosecond of computations.
     // So
     //   outcome.burnt_gas = wasm_insns(call) *
-    //       VMConfig::default().regular_op_cost
+    //       vm_config.regular_op_cost
     //   gas_cost_per_wasm_op = (measured * GAS_IN_MEASURE_UNIT *
-    //       VMConfig::default().regular_op_cost) /
+    //       vm_config.regular_op_cost) /
     //       (DIVISOR * NUM_ITERATIONS * outcome.burnt_gas)
     // Enough to return just
-    //    (measured * VMConfig::default().regular_op_cost) /
+    //    (measured * vm_config.regular_op_cost) /
     //       (outcome.burnt_gas * NUM_ITERATIONS),
     // as remaining can be computed with ratio_to_gas().
-    Ratio::new(
-        measured * (VMConfig::default().regular_op_cost as u64),
-        NUM_ITERATIONS * outcome.burnt_gas,
-    )
+    Ratio::new(measured * (vm_config.regular_op_cost as u64), NUM_ITERATIONS * outcome.burnt_gas)
 }
 
 type CompileCost = (u64, Ratio<u64>);
 
 fn compile(code: &[u8], gas_metric: GasMetric, vm_kind: VMKind) -> Option<CompileCost> {
     let start = start_count(gas_metric);
     for _ in 0..NUM_ITERATIONS {
-        let prepared_code = prepare::prepare_contract(code, &VMConfig::default()).unwrap();
+        let prepared_code = prepare::prepare_contract(code, &VMConfig::test()).unwrap();
```

Comment:
should you still use the 'test' version of the config here? (and not the one loaded from config_store)

Reply:
Strictly speaking, both are wrong. What we ideally should use in the estimator is unitary config -- a config where all costs are equal to `1` and just count the number of operations. Will hopefully do that after https://github.com/near/nearcore/pull/4733.

Reply:
@matklad Isn't it better to use runtime config with **last** protocol version to check that after protocol upgrades estimation still works and fees don't change unexpectedly?

Reply:
@matklad Isn't it better to use runtime config with **last** protocol version to check that estimation still works and fees don't change unexpectedly after protocol upgrades?

Reply:
:thinking: I think I am becoming convinced that that's actually might be indeed better. What I originally overlooked is that `VMConfig` contains not only fees, but also variou other runtime constants like `max_contract_size` and the like. While specific fees should be irrelevant for estimation, those other params might still be relevant.

No super strong opinion either way though.